### ISSUE TITLE
BETA pomodoro clock works but does not pass some of the tests

### ISSUE THREAD

**Issue by MarioLema** [freeCodeCamp/testable-projects-fcc#301]
#### Issue Description
I have constructed a Pomodoro clock following the requirements on the beta and although it is functional and it meets the requirements, the tester does not recognize them.

[Codepen Pomodoro](https://codepen.io/Ubim28/pen/KoGgLM)

The points that should pass are: 
(line references refer to the original Codepen file)
TIMER

**1.When I click the element with the id of reset, any running timer should be stopped, the value within id=“break-length” should return to 5, the value within id=“session-length” should return to 25, and the element with id=“time-left” should reset its default state.**

Line 146 selects the #reset id when clicked, variables are assigned the values requested on lines 152-153 and lines 159-162 pass them to the document. line 162 sets the timer to 25:00 with a formula that takes the minutes and sets it to that format. 

```
  $('#reset').click( () => {
    if(session === 'session'){
      clearInterval(counter);
    }else{
      clearInterval(breakCounter)
    };
    minutes = 25;
    breaks = 5;
    rawTime = minutes*60;
    rawBreak = breaks*60;
    buzz.pause();
    buzz.currentTime = 0;
    session = 'session';
    $('#timer-label').html(session);
    $('#session-length').html(minutes);
    $('#break-length').html(breaks);
    $('#time-left').html(clockTime(rawTime));
    event.preventDefault();
  })
```

**9.If the timer is running, the element with the id of time-left should display the remaining time in mm:ss format (decrementing by a value of 1 and updating the display every 1000ms).**

Tester declares that nothing is being displayed but you can clearly see the countdown going down.
Lines 40-76 declare the functions that run the timer and the timer for Breaks.

**12.When a session countdown reaches zero (NOTE: timer MUST reach 00:00), and a new countdown begins, the element with the id of timer-label should display a string indicating a break has begun.**

Same functions take care of this and you can see 'session' and 'break swapping when the countdowns swap.

**13.When a session countdown reaches zero (NOTE: timer MUST reach 00:00), a new break countdown should begin, counting down from the value currently displayed in the id=“break-length” element.**

Function timer (line 40) calls a new interval ( line 57 )  for the timerBreak function (line 62).

**14.When a break countdown reaches zero (NOTE: timer MUST reach 00:00), and a new countdown begins, the element with the id of timer-label should display a string indicating a session has begun.**

timerBreak function (line 62)  calls a new interval (line 74) to run again the timer function. 

**15.When a break countdown reaches zero (NOTE: timer MUST reach 00:00), a new session countdown should begin, counting down from the value currently displayed in the id=“session-length” element.**

The increment and decrement buttons( lines 112-127) set the variable minutes and display it, and the variable rawTime, total of seconds, each time they are pressed. 
You can still increment and decrement time when the counter is running and it will update the time immediately, but this is not explicitly forbidden on the instructions. When I tried to unbind the buttons or stop the user from using them when the countdown was running, even fewer tests would pass.

#### Browser Information

* Browser Name, Version:  Google Chrome Version 65.0.3325.181 (Official Build) (64-bit)
* Operating System: Windows (10 64bit)
* Mobile, Desktop, or Tablet: Desktop

**Comment by MarioLema** [comment]
I have made a different version of it where the timer with the minutes from the #session-length and #break-length only updates the timer with the first play(not after a pause) and after each interval.
It has not made any differences. Here is the link: [No instant timer update](https://codepen.io/Ubim28/pen/QmozyM?editors=0010)

**Comment by tchaffee** [comment]
@MarioLema Thank you for reporting this and we will look into this as soon as we can.

**Comment by tchaffee** [comment]
@MarioLema I might have found one problem. Look at this requirement:

 ```
I can see an element with corresponding id="time-left". NOTE: Paused or running, the value in this field should always be displayed in mm:ss format (i.e. 25:00).
```

`mm:ss` means you always need two places showing for the minutes. 

For example, with a time left of 4 minutes and 30 seconds the value in the "time-left" element should be `04:30`. Let us know what happens if you make that change.

**Comment by MarioLema** [comment]
I have changed the function clockTime (line24) to display the time with the extra 0 in front if the number passed is under 600(10 minutes = 600seconds). 

Unfortunately, it still does not pass any of the tests I referred to. [Pomodoro](https://codepen.io/Ubim28/pen/KoGgLM?editors=0010)

**Comment by tchaffee** [comment]
@MarioLema Thanks for giving that a try. I will take a closer look at this as soon as I can.

**Comment by tbushman** [comment]
@MarioLema I was able to get a few more tests to pass by adding `playing = false;` to your `$('#reset').click()` function (codepen line 145), while @tchaffee by adding a `resetTimer()` to the `before()` function in `/src/project-tests/pomodoro-clock-tests.js`, which I think can't hurt. I'll see if I can get more tests to pass.

**Comment by MarioLema** [comment]
@tbushman @tchaffee There was an if else statements on the `$('#reset').click()` to clear the right interval. I have removed it and now tests 1. of TIMER passes too. Less to go. Again I'll have to go over it too to see what I missed and make it more robust. Thank you

**Comment by MarioLema** [comment]
Well, I have managed to solve all of them after some work(trickery and witchcraft it seemed to me).  Both my timer and timerBreak functions would log the time AFTER they had deducted a second from the variable holding the seconds. With timerBreak (line 71 now) it was just a matter of writing the DOM before deducting but I had to use a different order for the timer one.

In any event, now it passes all the tests I thought it wouldn't pass. I don't know if I didn't read the error logs properly before or whatever @tchaffee changed made it give a different error log but that really helped me figure out (see "shooting in the dark") what was wrong.

Thank you both @tchaffee @tbushman.
